### ISSUE TITLE
IoD Performance 

### ISSUE THREAD

**Issue by eltonvivot** [telematics-lab/IoD_Sim#7]
Hello,

first, Happy New Year to everyone! =D

I did some tests running ns3 alone to understand it better and seems that IoD_Sim has a better performance in terms of simulation running speed. Did you make any changes that justify this improvement? I'm running ns3 scripts with 'waf'.

Best regards,
Elton.

**Comment by GiovanniGrieco** [comment]
Hello Elton,

thanks and happy year to you too!
When we write the code we always check if our approach is sufficient enough to run the simulation in a reasonable amount of time. But, frankly, we have never measured the performance improvements outside the scope of our main paper.

Could you please provide us some examples that you observed? It would be great to know: 
* What versions of IoD Sim have you tried.
* A copy of the logs and tested JSON configurations in an archive.

This would allow us to cross-check that we are effectively observing performance improvements, and thus confirm your hypothesis, or we are dealing with a bug somewhere.

Best regards

**Comment by eltonvivot** [comment]
Hello Giovanni,

Sorry for the late response. Due to unforeseen circumstances, I was not able to respond in a timely manner. Analyzing the versions of ns3 that I was using, I noticed that it was an older version, compared to the version used by IoD_sim. When I ran the same ns3 version used by IoD_Sim, I got very similar times.

I'm currently working on a fork of IoD_Sim and in the future I'll do more detailed performance tests. Once I have the results I can update you on my findings, but seems that the slower ns3 alone performance was due to its older version.

Again, sorry for the delay and any trouble it may have caused.

Cheers.
